libraries/SITL/SIM_RF_Ainstein_LR_D1.cpp

```diff
@@ -0,0 +1,57 @@
+/*
+   This program is free software: you can redistribute it and/or modify
+   it under the terms of the GNU General Public License as published by
+   the Free Software Foundation, either version 3 of the License, or
+   (at your option) any later version.
+
+   This program is distributed in the hope that it will be useful,
+   but WITHOUT ANY WARRANTY; without even the implied warranty of
+   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+   GNU General Public License for more details.
+
+   You should have received a copy of the GNU General Public License
+   along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+/*
+  Simulator for the Ainstein LR-D1 Serial rangefinder
+*/
+
+#include "SIM_RF_Ainstein_LR_D1.h"
+
+using namespace SITL;
+
+uint32_t RF_Ainstein_LR_D1::packet_for_alt(uint16_t alt_cm, uint8_t *buffer, uint8_t buflen)
+{
+    const uint8_t PACKET_LEN = 32;
+
+    if (buflen < PACKET_LEN) {
+        return 0;
+    }
+
+    uint8_t malfunction_alert = 0;
+
+    uint8_t snr;
+    if (alt_cm >= 65535) {
+        snr = 0;
+        alt_cm = 65535;
+    } else {
+        snr = 100;
+    }
```

Comment:
alt_cm can never be > 0xFFFF so might as well simply this

```suggestion
const uint8_t snr = (alt_cm == 0xFFFF) ? 0 : 100;
```